modular_skyrat/modules/SiliconQoL/code/robotic_factory.dm

```diff
@@ -0,0 +1,69 @@
+/obj/machinery/transformer_rp
+	name = "\improper Automatic Robotic Factory 5000"
+	desc = "A large metallic machine with an entrance and an exit. A sign on \
+		the side reads, 'Mass robot production facility'"
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "separator-AO1"
+	layer = ABOVE_ALL_MOB_LAYER // Overhead
+	density = TRUE
+	/// How many cyborgs are we storing
+	var/stored_cyborgs = 1
+	/// How many cyborgs can we store?
+	var/max_stored_cyborgs = 4
+	/// How much between the construction of a cyborg?
+	var/cooldown_duration = 3000 // 5 minutes
+	/// Handles the timer , shouldn't touch.
+	var/cooldown_timer
+	/// The countdown itself
+	var/obj/effect/countdown/transformer/countdown
+	/// The master AI , assigned when placed down with the ability.
+	var/mob/living/silicon/ai/masterAI
+
+/obj/machinery/transformer_rp/Initialize()
+	// On us
+	. = ..()
+	new /obj/machinery/conveyor/auto(loc, WEST)
+	countdown = new(src)
+	countdown.start()
+
+/obj/machinery/transformer_rp/examine(mob/user)
+	. = ..()
+	if(issilicon(user) || isobserver(user))
+		. += "It will create a new cyborg in [DisplayTimeText(cooldown_timer - world.time)]."
+
+/obj/machinery/transformer_rp/Destroy()
+	QDEL_NULL(countdown)
+	. = ..()
+
+/obj/machinery/transformer_rp/update_icon_state()
+	if(machine_stat & (BROKEN|NOPOWER))
+		icon_state = "separator-AO0"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/transformer_rp/attack_ghost(mob/dead/observer/target_ghost)
+	. = ..()
+	create_a_cyborg(target_ghost)
+
+/obj/machinery/transformer_rp/process()
```

Comment:
Perhaps give a ghost prompt once it's ready for a new cyborg?

Reply:
added it now , accidentally deleted it by accident.